example-app/src/store.ts

```diff
@@ -1,24 +1,30 @@
-import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
-import { createPendingMiddleware } from 'redux-pending-effects';
-import createSagaMiddleware from '@redux-saga/core';
-
-import { rootReducer as reducer } from './reducers/rootReducer';
-import { rootSaga } from './sagas';
-
-const rpeMiddlewares = createPendingMiddleware({
-  promiseMiddleware: true,
-  toolkitMiddleware: true
-});
-export const sagaMiddleware = createSagaMiddleware();
-export const defaultMiddlewares = getDefaultMiddleware({
-  serializableCheck: false
-});
-export const middleware = [
-  ...rpeMiddlewares,
-  sagaMiddleware,
-  ...defaultMiddlewares
-];
-
-export const store = configureStore({ reducer, middleware });
-
-sagaMiddleware.run(rootSaga);
+import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
+import {
+  createPendingMiddleware,
+  pendingSagaMiddleware
+} from 'redux-pending-effects';
+import createSagaMiddleware from '@redux-saga/core';
+
+import { rootReducer as reducer } from './reducers/rootReducer';
+import { rootSaga } from './sagas';
+
+const rpeMiddlewares = createPendingMiddleware({
+  promiseMiddleware: true,
+  toolkitMiddleware: true
+});
+export const sagaMiddlewareOptions = {
+  effectMiddlewares: [pendingSagaMiddleware]

```

Comment:
I already wrote a private message to you, but to be transparent I left the comment

great idea but looks ugly at the moment
can we come up with a more concise API?